File: python/849. Maximize Distance to Closest Person.py
class Solution(object):
    def maxDistToClosest(self, seats):
        """
        :type seats: List[int]
        :rtype: int
        """
        max_len = 0
        s = 0
        l = 0
        i = 0
        n = len(seats)
        while i < n and seats[i] != 1:
            i += 1
        l = i

        while i < n:
            if seats[i]==1:
                s = 0
            else:
                s += 1
            max_len = max(max_len, s)
            i += 1
        if max_len&1:
            max_len = max_len//2 + 1
        else:
            max_len = max_len//2
        return max(max_len, s, l)

        

File: go/src/leetcode/849/849. Maximize Distance to Closest Person.go
package main

/*
*
https://leetcode.cn/problems/maximize-distance-to-closest-person/description/
*/
func maxDistToClosest(seats []int) int {
	var maxLen, l, s, i int
	n := len(seats)
	for ; i < n && seats[i] != 1; i++ {
	}
	l = i
	for ; i < n; i++ {
		if seats[i] == 1 {
			s = 0
		} else {
			s++
		}
		maxLen = max(maxLen, s)
	}
	if maxLen&1 == 1 {
		maxLen = maxLen>>1 + 1
	} else {
		maxLen = maxLen >> 1
	}
	return max(maxLen, max(s, l))
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}
